pkg/credentialprovider/provider.go

```diff
@@ -121,3 +121,24 @@ func (d *CachingDockerConfigProvider) Provide() DockerConfig {
 	d.expiration = time.Now().Add(d.Lifetime)
 	return d.cacheDockerConfig
 }
+
+// Reset expiration so that the provider will get configure from real provider instead of cache.
+// This is useful in testings.
+func (d *CachingDockerConfigProvider) resetExpiration() {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	d.expiration = time.Now().Add(-1 * time.Hour)
+}
+
+// ResetDefaultDockerProviderExpiration resets .dockercfg provider's expiration.
```

Comment:
Same here. This function is not used outside testing...

Reply:
@yujuhong yep, but the timeout is set at init(), there is no easy way to set it outside the package. Any suggestions of a gentle way?

Reply:
I don't think dockershim should be testing the internals of the `CachingDockerConfigProvider`.
One option is to decouple the testing of credential provider by mocking it in dockershim. The other is to export `defaultDockerConfigProvider` and use that.

Reply:
ok, will remove this test case.

Reply:
we may add a docker config test in `pkg/credentialprovider` package later.